packages/flutter/lib/src/material/input_border.dart

```diff
@@ -432,7 +431,7 @@ class OutlineInputBorder extends InputBorder {
     const double trCornerArcSweep = cornerArcSweep;
     if (start + extent < scaledRRect.width - scaledRRect.trRadiusX) {
       path
-        ..relativeMoveTo(extent, 0.0)
+        ..moveTo(scaledRRect.left + start + extent, scaledRRect.top)
```

Comment:
This change and the previous changes near line 425 make the gap the correct width.  Before, it was too wide.